src/Database/Redis/Cluster.hs

```diff
@@ -388,22 +375,36 @@ evaluateTransactionPipeline shardMapVar refreshShardmapAction conn requests' = d
     -- make arbitrary decisions about how long to paus before the retry and how
     -- often to retry, so instead we'll propagate the error to the library user
     -- and let them decide how they would like to handle the error.
-    when (any moved resps)
-      (hasLocked $ modifyMVar_ shardMapVar (const refreshShardmapAction))
-    retriedResps <- retryBatch shardMapVar refreshShardmapAction conn 0 requests resps
-    return retriedResps
-
-nodeConnForHashSlot :: Exception e => MVar ShardMap -> Connection -> e -> HashSlot -> IO NodeConnection
-nodeConnForHashSlot shardMapVar conn exception hashSlot = do
-    let (Connection nodeConnsMvar _ _ _ _) = conn
-    nodeConns <- hasLocked $ readMVar nodeConnsMvar
-    (ShardMap shardMap) <- hasLocked $ readMVar shardMapVar
+    loopM (\case 
```

Comment:
We can get multiple moved in multi exec right, there is a chance that `requestNode newNodeConn requests` can get called multiple times